Assets/Scripts/KatamariMovement.cs

```diff
@@ -50,24 +64,109 @@ void Update()
         verticalInput = Input.GetAxis("Vertical");
         hitInput = Input.GetAxis("Jump");
         stopInput = Input.GetAxis("Stop");
-        isGolfHitMode = Input.GetAxis("Mode") > 0.5;
 
-        hitXAngle += hitXAngleSpeed * verticalInput * Time.deltaTime;
+
+
+
+        // turn the power bar on and off depending if we can hit hit the ball
+        if (power_bar_active)
+        {
+            power_bar.SetActive(true);
+            isGolfHitMode = true;
+            this.GetComponent<LineRenderer>().enabled = true;
+
+
+        }
+        else
+        {
+            power_bar.SetActive(false);
+            this.GetComponent<LineRenderer>().enabled = false;
+        }
+
+        if (resourceManager.can_hit())
+        {
+            power_bar_active = true;
+        }
+
+
+        // SLOW MOTION
+
+        if (power_bar_active == false && hitInput > 0.5 && resourceManager.GetComponent<ResourceManager>().GetTimeSinceLastHit() >= 0.2f)
+        {
+            time_manager.SlowMotion();
+        }
+
+
+        else
+        {
+            time_manager.NormalTime();
+
+        }
+
+
+
+
+        if (power_bar_active == false && hitInput > 0.5 && resourceManager.GetComponent<ResourceManager>().GetTimeSinceLastHit() <= 0.2f)
```

Comment:
To stay consistent with the above:
```suggestion
        if (power_bar_active == false && hitInput > 0.5 && isFlyMovementDeadTime)
```